bin/strata-cli/src/seed/password.rs

```diff
@@ -29,6 +29,14 @@ impl HashVersion {
 }
 
 impl Password {
+    /// Sets a password from a string. (The complexity of the password is not checked.)
+    pub fn new(password: &str) -> Self {
+        Self {
+            inner: password.to_string(),
```

Comment:
This could take `String` directly right?  Also this would be "constructs" rather than "sets" couldn't it?

Reply:
It certainly could, but I wanted to make it more general and not require heap allocation by the caller. Thoughts on this?

Good call on the docstrings. Reading it again now, the current wording feels awkward.